mozetl/search/daily_search_rollups.py

```diff
@@ -1,309 +1,371 @@
+        "distribution_id": "MOZILLA",
+        "search_count": 0,
+    }
+
+    dataframe = main_summary.select(columns)
+    exploded = explode_searches(dataframe).na.fill(defaults)
+
+    # don't include shares if we process the monthly rollup
+    if mode == "monthly":
+        processed = exploded
+    else:
+        processed = with_shares(exploded)
+    search_rollup = rollup_searches(processed, attributes, mode)
+
+    return search_rollup
+
+
+def format_spark_path(bucket, prefix):
+    return "s3://{}/{}".format(bucket, prefix)
+
+
+def get_end_date(ds_start, period):
+    """ Return the end date given the start date and period. """
+    date_start = arrow.get(ds_start, "YYYYMMDD")
+    if period == "monthly":
+        date_end = date_start.replace(months=+1)
+    else:
+        date_end = date_start.replace(days=+1)
+    ds_end = date_end.format("YYYYMMDD")
+
+    return ds_end
+
+
+def extract(spark, path, ds_start, period):
+    """Extract the source dataframe from the spark compatible path.
+
+    spark: SparkSession
+    path: path to parquet files in s3
+    ds_start: inclusive date
+    """
+    ds_end = get_end_date(ds_start, period)
+
+    return (
+        spark.read
+        .option("mergeSchema", "true")
+        .parquet(path)
+        .where((F.col("submission_date_s3") >= ds_start) &
+               (F.col("submission_date_s3") < ds_end))
+    )
+
+
+def get_last_manifest_version(bucket, prefix, pattern):
+    """ Get the version of the last manifest with the same pattern."""
+    s3 = boto3.resource('s3')
+    bucket_obj = s3.Bucket(bucket)
+
+    last_manifest_version = None
+    for obj in bucket_obj.objects.filter(Prefix=prefix):
+        # match for items that match the current pattern
+        if re.search(pattern, obj.key):
+            # capture the current version number of this pattern
+            search_version = re.search("-v(.*?)\.txt", obj.key)
+            if search_version and search_version.group(1):
+                # get the captured group
+                version = int(search_version.group(1))
+                last_manifest_version = max(last_manifest_version, version)
+
+    return last_manifest_version
+
+
+def get_csv_locations(bucket, prefix):
+    """Return the locations of csv files within a prefix. The final locations
+    are formatted path strings that can be read by spark.
+    """
+    s3 = boto3.resource('s3')
+    bucket_obj = s3.Bucket(bucket)
+
+    csv_files = []
+    for obj in bucket_obj.objects.filter(Prefix=prefix):
+        if obj.key.endswith(".csv"):
+            csv_files.append("s3://{}/{}".format(bucket, obj.key))
+    return csv_files
+
+
+def write_manifest(bucket, prefix, mode, version, start_ds, csv_paths, retry_max=10):
+    """ Write a manifest file with the location of the daily rollup files.
+
+    A manifest enumerates the locations of csv files that contain the rollup.
+    This might be useful if the file is partitioned across multiple files.
```

Comment:
Please add a comment indicating that the manifest is needed as part of the subsequent Vertica integration per [Sam's comment](https://github.com/mozilla/python_mozetl/pull/52/files#r122309470)